### PR TITLE
Improve documentation for excess spikes error in `FrameSliceSorting`.

### PR DESCRIPTION
**Pull request by h-mayorquin**
From the discussion here:
https://github.com/SpikeInterface/spikeinterface/issues/2573#issuecomment-1995993813

I was kind of recluctanct to add "could indicate a bug on the sorter" as this is a very general function but I guess a sorting object can only come from sorter so...

### CODE REVIEW

**Comment by zm711** [comment]
I think you're right to be a little wary. The only thing I could think of is maybe a rounding error. Like if the sampling_rate is 30k, but because we store it as a float it ends up being 30_000.000002434 and so when we multiply that by the real big sample numbers it leads to some error in certain sorters. But I guess we would have to track down these failures and see if they are within a few samples of the end of the recording or if they seem to have random times. Other than a rounding error everything else would point to an issue with the sorter, no?

**Comment by alejoe91** [comment]
Yeah it must be a problem with the sorter!

File: src/spikeinterface/core/frameslicesorting.py [hunk 1]
```diff
@@ -56,8 +56,9 @@ def __init__(self, parent_sorting, start_frame=None, end_frame=None, check_spike
             ), "`start_frame` should be smaller than the sortings' total number of samples."
             if check_spike_frames and has_exceeding_spikes(parent_sorting._recording, parent_sorting):
                 raise ValueError(
-                    "The sorting object has spikes exceeding the recording duration. You have to remove those spikes "
-                    "with the `spikeinterface.curation.remove_excess_spikes()` function"
+                    "The sorting object has spikes whose times go beyond the recording duration."
+                    "This could indicate a bug in the sorter. Double check the time of both objects"
```
[the remark this answers is not in the document]
Comment (zm711) [reply src/spikeinterface/core/frameslicesorting.py:60]: That's a great point Alessio!

[the remark this answers is not in the document]
Comment (h-mayorquin) [reply src/spikeinterface/core/frameslicesorting.py:60]: Fair enough, feel free to commit this change.

What I really wish we had is a way of printing the problematic timestamps and the recording duration so the user knows at glance the nature of the problem.

But the problem is that the function that does the check returns a boolean and some small refactor would be necessary. That might not be as quick as this improvement. 

I might do that later when I find  some time.

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**